### DISCUSSION TITLE
Interpolating temperature field from input data

### DISCUSSION THREAD

**Discussion by chakra34** [idaholab/moose#24869] [Q&A General] [answered]
Hi,

So I have a temperature field data, that has X, Y, Z, Temp values. When I read the file via PiecewiseConstantCSV I get stepped results which makes me think that the interpolation between them is not that great. So I was wondering are there any different such interpolation strategies in-built in MOOSE that I can play around with ? 

I was looking at PiecewiseMultilinear but it seems it needs ascending order of AXIS values-- (or is my interpretation wrong ?). In other words, can such PiecewiseMultilinear be used to better interpolate such 3D data set into my grid ?

Thanks,

Best,
Aritra

**Comment by GiudGiud** [comment, marked as answer]
Hello

if you have general 3D data, so not on a grid, the PiecewiseConstantFromCSV is the only option. 

However it really would not be hard to build a linear version instead of the constant step interpolation 

Guillaume

**Reply by chakra34** [reply]
got it thanks :)
